src/coordinator/api/v1/handler/database/create.go

```diff
@@ -183,61 +214,78 @@ func (h *createHandler) parseRequest(r *http.Request) (*admin.NamespaceAddReques
 }
 
 func defaultedNamespaceAddRequest(r *admin.DatabaseCreateRequest) (*admin.NamespaceAddRequest, error) {
-	options := dbnamespace.NewOptions()
+	opts := dbnamespace.NewOptions()
 
 	switch dbType(r.Type) {
 	case dbTypeLocal, dbTypeCluster:
-		options.SetRepairEnabled(false)
+		opts = opts.SetRepairEnabled(false)
+		retentionOpts := opts.RetentionOptions()
 
 		if r.RetentionPeriodNanos <= 0 {
-			options.RetentionOptions().SetRetentionPeriod(defaultLocalRetentionPeriod)
+			retentionOpts = retentionOpts.SetRetentionPeriod(defaultLocalRetentionPeriod)
 		} else {
-			options.RetentionOptions().SetRetentionPeriod(time.Duration(r.RetentionPeriodNanos))
+			retentionOpts = retentionOpts.SetRetentionPeriod(time.Duration(r.RetentionPeriodNanos))
 		}
 
-		retentionPeriod := options.RetentionOptions().RetentionPeriod()
+		retentionPeriod := retentionOpts.RetentionPeriod()
 		if r.ExpectedSeriesDatapointsPerHour > 0 {
 			blockSize := time.Duration(blockSizeFromExpectedSeriesScalar / r.ExpectedSeriesDatapointsPerHour)
-			if blockSize < minBlockSize {
-				blockSize = minBlockSize
-			} else if blockSize > maxBlockSize {
-				blockSize = maxBlockSize
+			if blockSize < minRecommendCalculateBlockSize {
```

Comment:
The way this should be read is, if the "calculated block size" (here `blockSize`) is less than the "minimum to recommend when calculating block size" (here `minRecommendCalculateBlockSize`) then just set it to the "minimum to recommend when calculating block size" (i.e. `minRecommendCalculateBlockSize`).

Does that make sense?  So it is recommended, its just the minimum we would recommend when making recommendation based on calculating block sizes when specifying expected datapoints per block.

Reply:
Yeah, just it's a bit weird calling it `Recommended` since there's an [implication](https://res.cloudinary.com/teepublic/image/private/s--FO3xh14a--/t_Preview/b_rgb:0195c3,c_limit,f_jpg,h_630,q_90,w_630/v1491907970/production/designs/1425101_1.jpg) that you can ignore the recommendation and override it?

Might be getting too pedantic about it though, take it or leave it

Reply:
You can have smaller block sizes, you just have to specify it explicitly.  Well actually no that's not true, I'll add it to the protobuf so you can use either the expected datapoints per block or an explicit block size.